packages/doenetml-worker-rust/doenetml-core/src/component/text.rs

```diff
@@ -17,4 +32,264 @@ pub struct Text {
     pub descendant_names: HashMap<String, Vec<ComponentIdx>>,
 
     pub position: Option<DastPosition>,
+
+    pub state_variables: Vec<StateVar>,
+
+    pub rendered_state_variable_indices: Vec<usize>,
+
+    pub public_state_variable_indices: Vec<usize>,
+
+    pub state_variable_name_to_index: HashMap<String, usize>,
+
+    pub value_state_var_view: StateVarReadOnlyViewTyped<String>,
+
+    pub component_profile_state_variables: Vec<ComponentProfileStateVariable>,
```

Comment:
There seems to be a lot of boiler plate here. Can a generic struct be created to house all the non-text-specific stuff?

Reply:
I couldn't figure out how to do that in rust. I removed most of the boilerplate code with the derive macros, but couldn't figure out how to remove the boilerplate specification of the fields. The derive macros don't let you change the structure itself; they only let you add code after the struct.

I decided to leave further attempts for removing the boilerplate fields to our future selves, but if you have any ideas on how to fix it now, all the better!

Reply:
It looks like one could add common fields with attribute macros. Trying to decide if it is worth it now.

Reply:
Added an  `add_standard_component_fields` attribute macro to put all those standard fields in the components.

That gets rid of the boilerplate fields. I hope all these macros aren't too confusing to future developers.

Reply:
Removed macro